pkg/cmd/root_test.go

```diff
@@ -15,220 +15,187 @@
-	})
-
-	It("should prioritize '--manifest' command line option over 'GETENVOY_MANIFEST_URL' environment variable", func() {
-		unexpected := "https://host/path/that/should/be/ignored" //nolint:goconst
-		expected := "https://host/path/to/manifest"
-
-		By("running command")
-		os.Setenv("GETENVOY_MANIFEST_URL", unexpected)
-		c := newRootCmd(stdout, stderr)
-		c.SetArgs([]string{"--manifest", expected, "fake-command"})
-		err := cmdutil.Execute(c)
-		Expect(err).NotTo(HaveOccurred())
-
-		By("verifying global state")
-		Expect(manifest.GetURL()).To(Equal(expected))
-	})
-
-	It("should reject empty '--manifest' command line option", func() {
-		unexpected := "https://host/path/that/should/be/ignored" //nolint:goconst
-
-		By("running command")
-		os.Setenv("GETENVOY_MANIFEST_URL", unexpected)
-		c := newRootCmd(stdout, stderr)
-		c.SetArgs([]string{"--manifest=", "fake-command"})
-		err := cmdutil.Execute(c)
-		Expect(err).To(HaveOccurred())
-
-		By("verifying command output")
-		Expect(stdout.String()).To(BeEmpty())
-		Expect(stderr.String()).To(Equal(`Error: GetEnvoy manifest URL cannot be empty
-
-Run 'getenvoy fake-command --help' for usage.
-`))
-	})
-
-	It("should reject invalid '--manifest' command line option", func() {
-		unexpected := "https://host/path/that/should/be/ignored" //nolint:goconst
-		invalid := "/not/a/url"
-
-		By("running command")
-		os.Setenv("GETENVOY_MANIFEST_URL", unexpected)
-		c := newRootCmd(stdout, stderr)
-		c.SetArgs([]string{"--manifest", invalid, "fake-command"})
-		err := cmdutil.Execute(c)
-		Expect(err).To(HaveOccurred())
-
-		By("verifying command output")
-		Expect(stdout.String()).To(BeEmpty())
-		Expect(stderr.String()).To(Equal(`Error: "/not/a/url" is not a valid manifest URL
-
-Run 'getenvoy fake-command --help' for usage.
-`))
-	})
-})
+	tests := []testCase{ // we don't test default as that depends on the runtime env
+		{
+			name:     "default is https://tetrate.bintray.com/getenvoy/manifest.json",
+			setup:    emptySetup,
+			expected: "https://tetrate.bintray.com/getenvoy/manifest.json",
+		},
+		{
+			name: "GETENVOY_MANIFEST_URL env",
+			setup: func() func() {
+				return RequireSetenv(t, "GETENVOY_MANIFEST_URL", "http://GETENVOY_MANIFEST_URL/env")
+			},
+			expected: "http://GETENVOY_MANIFEST_URL/env",
+		},
+		{
+			name:     "--manifest arg",
+			args:     []string{"--manifest", "http://manifest/arg"},
+			setup:    emptySetup,
+			expected: "http://manifest/arg",
+		},
+		{
+			name: "prioritizes --manifest arg over GETENVOY_MANIFEST_URL env",
+			args: []string{"--manifest", "http://manifest/arg"},
+			setup: func() func() {
+				return RequireSetenv(t, "GETENVOY_MANIFEST_URL", "http://GETENVOY_MANIFEST_URL/env")
+			},
+			expected: "http://manifest/arg",
+		},
+	}
+
+	for _, test := range tests {
+		test := test // pin! see https://github.com/kyoh86/scopelint for why
+
+		t.Run(test.name, func(t *testing.T) {
+			tearDown := test.setup()
+			defer tearDown()
```

Comment:
Shall we check if setup func is not nil, then we could remove `emptySetup`?
```suggestion
		if test.setup != nil {
			tearDown := test.setup()
			defer tearDown()
		}
```

Reply:
good! I didn't commit this as it drifted your intent, but thanks!